service/state/rpc.go

```diff
@@ -0,0 +1,114 @@
+package state
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/gorilla/mux"
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/celestiaorg/celestia-node/node/rpc"
+)
+
+var log = logging.Logger("state/rpc")
+
+var (
+	addrKey = "address"
+	txKey   = "tx"
+)
+
+func (s *Service) RegisterEndpoints(rpc *rpc.Server) {
+	rpc.RegisterHandlerFunc("/balance", s.handleBalanceRequest, http.MethodGet)
+	rpc.RegisterHandlerFunc(fmt.Sprintf("/balance/{%s}", addrKey), s.handleBalanceForAddrRequest, http.MethodPost)
+	rpc.RegisterHandlerFunc(fmt.Sprintf("/submit_tx/{%s}", txKey), s.handleSubmitTx, http.MethodPost)
+}
+
+func (s *Service) handleBalanceRequest(w http.ResponseWriter, r *http.Request) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	bal, err := s.accessor.Balance(ctx)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		log.Errorw("serving /balance request", "err", err)
+		return
+	}
+	resp, err := bal.Marshal()
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		log.Errorw("serving /balance request", "err", err)
+		return
+	}
+	w.WriteHeader(http.StatusOK) // TODO @renaynay set header before or after writing resp?
```

Comment:
You can skip calling `w.WriteHeader` with `http.StatusOK`, bc it will be called internally in `w.Write(resp)`